modules/new_serverpod_auth/serverpod_auth_core/serverpod_auth_core_server/lib/src/jwt/business/jwt_util.dart

```
@@ -173,16 +174,18 @@ class JwtUtil {
         issuer: _issuer,
       );
     } catch (_) {
-      final fallbackAlgorithm = _fallbackVerificationAlgorithm;
-      if (fallbackAlgorithm == null) {
-        rethrow;
+      for (final fallbackAlgorithm in _fallbackVerificationAlgorithms) {
```

Suggestion: Combine this into a single list that is validated with the same rules. We should rethrow the first error that occurs if none of them match.